src/other/PropertyEditor.js

```diff
@@ -1,401 +1,266 @@
+                element.selectAll(context._tablePath+" [data-widgetid] tr").style("display","none");
+                var selected = selectedItems.map(function(n){return n.id();});
+                if(selected.length > 0){
+                    var selectString = selected.map(function(n){
+                                return '[data-widgetid="'+n+'"] tr';
+                            }).join(",");
+                    element.selectAll(context._tablePath+" > tbody > tr.property-widget-wrapper").style("display","table-row");
+                    element.selectAll(selectString).style("display","table-row");
+                    context.hideNonWidgets(true);
+                } else {
+                    element.selectAll(context._tablePath+" tr").style("display","table-row");
+                    context.hideNonWidgets(false);
                 }
-                break;
+                context.render();
+            };
         }
-        Accordion.prototype.update.apply(this, arguments);
     };
     
     PropertyEditor.prototype.exit = function (domNode, element) {
-        this.removeWatches();
         HTMLWidget.prototype.exit.apply(this, arguments);
     };
     
-    PropertyEditor.prototype.removeWatches = function () {
-        this.__propertyEditor_watch.forEach(function (watchObj) {
-            watchObj.remove();
-        });
-        this.__propertyEditor_watch = [];
-    };
-    
-    PropertyEditor.prototype.createWatch = function(obj){
+    PropertyEditor.prototype.exitAllChildren = function () {
         var context = this;
-        this.__propertyEditor_watch.push(obj.watch(function(key, newVal){
-            if(typeof (newVal) === "object"){
-                context.render();
+        for(var paramId in context._flatParams){
+            if(context._flatParams[paramId] instanceof Array){
+                var widgetIdArr = context._flatParams[paramId];
+                for(var i = 0;paramId < widgetIdArr.length;paramId++){
+                    context._childWidgets[widgetIdArr[i]].watch.remove();
+                    context._childWidgets[widgetIdArr[i]].target(null);
+                    context._childWidgets[widgetIdArr[i]].exitAllChildren();
+                    delete context._childWidgets[widgetIdArr[i]];
+                }
             }
-        }));
+        }
     };
     
-    PropertyEditor.prototype.categorizeParams = function (arr) {
-        var categories = {
-            "Colors":["palette","color"],
-            "Font":["font"],
-            "Axis":["axis"],
-            "General":[],
-        };
-        var ret = {};
-        for(var p in arr){
-            for(var c in categories){
-                if(p.toLowerCase().indexOf(categories[c]) !== -1){
-                    break;
+    PropertyEditor.prototype.setWatches = function (domNode, element) {
+        var context = this;
+        context.widget().watch(function(paramId,newVal,oldVal){
+            if(context._flatParams[paramId] instanceof Array){
```

Comment:
I am hoping that once your not leaking watches it will be fast enough to re-render when any value changes?